src/Application.vala

```diff
@@ -363,6 +371,65 @@ public class ValaLint.Application : GLib.Application {
         return num_errors > 0;
     }
 
+    bool print_mistakes_json (Vala.ArrayList<FileData?> file_data_list) {
+        int num_errors = 0;
+        int num_warnings = 0;
+        bool first_mistake = true;
+        application_command_line.print ("{\"mistakes\": [");
+
+        foreach (FileData file_data in file_data_list) {
+            if (!file_data.mistakes.is_empty) {
+                foreach (FormatMistake mistake in file_data.mistakes) {
+                    if (first_mistake) {
+                        first_mistake = false;
+                    } else {
+                        application_command_line.print (",");
+                    }
+                    switch (mistake.check.state) {
+                        case ERROR:
+                            num_errors++;
+                            break;
+
+                        case WARN:
+                            num_warnings++;
+                            break;
+
+                        default:
+                            break;
+                    }
+
+                    string mistakes_end = "";
+                    if (print_mistakes_end) {
+                        mistakes_end = "\"endLine\":%i,\"endColumn\":%i,".printf (mistake.end.line, mistake.end.column);
+                    }
+
+                    application_command_line.print (
+                        "{\"filename\":\"%s\"," +
+                        "\"line\":%i," +
+                        "\"column\":%i,%s" +
+                        "\"level\":\"%s\"," +
+                        "\"message\":\"%s\"," +
+                        "\"ruleId\":\"%s\"}",
```

Comment:
This string was over 120 characters so I split it up. Not sure what preferred style here is, maybe the `+`s should be moved to start of the next line and indent every line after the first? Not sure what you'd prefer.

Reply:
Marking as resolved since the code in question is no longer present.